protoc-gen-swagger/genswagger/template.go

```diff
@@ -1187,3 +1230,70 @@ func extractSwaggerOptionFromFileDescriptor(file *pbdescriptor.FileDescriptorPro
 	}
 	return opts, nil
 }
+
+func swaggerSchemaFromProtoSchema(s *swagger_options.Schema, reg *descriptor.Registry) swaggerSchemaObject {
+	ret := swaggerSchemaObject{
+		schemaCore:   schemaCore{},
+		ExternalDocs: protoExternalDocumentationToSwaggerExternalDocumentation(s.GetExternalDocs()),
+		// TODO(johanbrandhorst): Add more fields?
+	}
+	if s.GetJsonSchema().GetRef() != "" {
+		swaggerName := fullyQualifiedNameToSwaggerName(s.GetJsonSchema().GetRef(), reg)
+		if swaggerName != "" {
+			ret.schemaCore.Ref = "#/definitions/" + swaggerName
+		} else {
+			ret.schemaCore.Ref += s.GetJsonSchema().GetRef()
+		}
+	} else {
```

Comment:
One thing this doesn't do is include external messages that aren't already inlined. I think we could potentially do some proto registry lookups and add those messages if necessary, but it'd be quite complex so I'm not sure we want to?